### ISSUE TITLE
Failed to convert object detection model to tensorflow lite

### ISSUE THREAD

**Issue by valorusa** [tensorflow/tensorflow#37761]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 18.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: from binary 
- **TensorFlow version (use command below)**: 2.0
- **Python version**: 3.69
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:7.4.0
- **CUDA/cuDNN version**: 10.2
- **GPU model and memory**:
- **Exact command to reproduce**:
frozen_model_path = '/home/tflite/tflite_graph.pb'
input_arrays=["image_tensor"]
output_arrays=["detection_boxes","detection_scores","num_detections","detection_classes"]
input_tensor={"image_tensor":[1,300,300,3]}
converter = tf.compat.v1.lite.TFLiteConverter.from_frozen_graph(frozen_model_path, input_arrays, output_arrays, input_tensor)
tflite_model = converter.convert()

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
I am using object detection API. I have a pretrained ssd_mobilenet_v2_coco model and I want to convert it to tensorflow lite(and do float16 quantization). In the folder of the model, I have checkpoint, frozen_inference_graph.pb, model.ckpt.data-00000-of-00001, model.ckpt.index, model.ckpt.meta, pipeline.config, and saved_model folder.

I used the following command to get the tflite.pb
python3 export_tflite_ssd_graph.py --output_directory=/home/tflite --pipeline_config_path=/home/exportedLP1/pipeline.config --trained_checkpoint_prefix=/home/exportedLP1/model.ckpt

Then I used the method posted online to convert tflite.pb. The last step  tflite = converter.convert() caused "abort(core dumped)'. According to posts I read online, I should be able to convertt to tflite without any issue.

### Source code / logs
Current thread 0x00007efbff6f1740 (most recent call first):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/lite/toco/python/toco_from_protos.py", line 56 in execute
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 250 in _run_main
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 299 in run
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/platform/app.py", line 40 in run
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/lite/toco/python/toco_from_protos.py", line 93 in main
  File "/usr/local/bin/toco_from_protos", line 8 in <module>
Aborted (core dumped)

**Comment by Saduf2019** [comment]
@valorusa 
please share simple stand alone code for us to replicate the issue faced by you, with complete error logs.

**Comment by valorusa** [comment]
Hi, all the codes are here:
 frozen_model_path = '/home/tflite/tflite_graph.pb'
input_arrays=["image_tensor"]
output_arrays=["detection_boxes","detection_scores","num_detections","detection_classes"]
input_tensor={"image_tensor":[1,300,300,3]}
converter = tf.compat.v1.lite.TFLiteConverter.from_frozen_graph(frozen_model_path, input_arrays, output_arrays, input_tensor)
tflite_model = converter.convert()

**Comment by Saduf2019** [comment]
@valorusa 
i have replicated the code shared by you and [face this error](https://colab.sandbox.google.com/gist/Saduf2019/d7086f73cd08b759c26f24c42189d82f/untitled106.ipynb), as requested please share simple stand alone code such that we can replicate the error faced by you to analyse it.

**Comment by valorusa** [comment]
Hi Thanks for the response. I attached the file /home/tflite/tflite_graph.pb here. Just unzip it to /home/tflite/

[tflite_graph.zip](https://github.com/tensorflow/tensorflow/files/4375693/tflite_graph.zip)

**Comment by valorusa** [comment]
Hi Thanks for the response. I tried to attached the file /home/tflite/tflite_graph.pb here. But it is too large to attach it here. So I add the description here:
1.go to tensorflow object detection model zoo to download **ssd_mobilenet_v2_coco**. unzip it under current directory you will get home/your_usrname/ssd_mobilenet_v2_coco_2018_03_29/, and open the  pipeline.config file and comment out this line _batch_norm_trainable: true_
2.make a new directory under your user name: /home/your_usrname/tflite, 
3.open the object detection api directory, go to ./models/research/object_detection 
4.then in the terminal, run  python3 export_tflite_ssd_graph.py  --output_directory=/home/your_usrname/tflite --pipeline_config_path=/home/your_usrname/ssd_mobilenet_v2_coco_2018_03_29/pipeline.config --trained_checkpoint_prefix=/home/your_username/ssd_mobilenet_v2_coco_2018_03_29/model.ckpt
5.now you should be able to find tflite_graph.pb and tflite_graph.pbtxt in home/your_usrname/tflite/
6.in python, run the following codes:
  frozen_model_path = '/home/your_usrname/tflite/tflite_graph.pb'
  input_arrays=["image_tensor"]
  output_arrays=["detection_boxes","detection_scores","num_detections","detection_classes"]
  input_tensor={"image_tensor":[1,300,300,3]}
  converter = tf.compat.v1.lite.TFLiteConverter.from_frozen_graph(frozen_model_path, input_arrays, output_arrays, input_tensor)
  tflite_model = converter.convert()

**Comment by Chris1hub** [comment]
Hello @valorusa I found your description very useful. Just one question: 
Where did you find information about input_arrays and output_arrays? 
I successfully converted ssd_mobilenet_v2_coco_2018_03_29 but I am not able to convert a custom model because I don't know input and output arrays.
Thanks

**Comment by valorusa** [comment]
Hi @Chris1hub I use the tensor flow object detection api to do the object detection and recognition. The input arrays are the names of the input tensors and the output arrays are the names of the output tensors. They are used in the below  inference routine:

with session.graph.as_default():
        image_tensor = detect_graph.get_tensor_by_name('image_tensor:0')
        
        # Each box represents a part of the image where a particular object was detected.
        boxes = detect_graph.get_tensor_by_name('detection_boxes:0')
        
        # Each score represent how level of confidence for each of the objects.
        # Score is shown on the result image, together with the class label.
        scores = detect_graph.get_tensor_by_name('detection_scores:0')
        classes = detect_graph.get_tensor_by_name('detection_classes:0')
        num_detections = detect_graph.get_tensor_by_name('num_detections:0')

        # Actual detection.
        (boxes, scores, classes, num_detections) = session.run([boxes, scores, classes, num_detections],feed_dict={image_tensor: image_np_expanded})
        return (boxes, scores, classes, num_detections)

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.

**Comment by sushreebarsa** [comment]
@valorusa Could you please try on latest stable version of tf 2.5 or 2.4.1 and let us know if this is still an issue.Thanks!
